src/components/FlagsMenu/FlagsMenu.tsx

```diff
@@ -29,6 +34,32 @@ const FlagsMenu = (props: FlagsMenuProps) => {
     ...rest
   } = props
 
+  // Idem for the translations
+  const displayNames = React.useMemo(() => {
+    return getDisplayNames(langOfCountryName)
+  }, [langOfCountryName])
+
+  // Don't need to refilter when the list is already displayed
+  const countriesFiltered = React.useMemo(() => {
+    const ISO_CODES_SORTED = sortAlphabeticallyCountryCodes(
+      ISO_CODES,
+      displayNames
+    )
+
+    return filterCountries(ISO_CODES_SORTED, {
+      onlyCountries,
+      excludedCountries,
+      continents,
+      preferredCountries
+    })
+  }, [
+    continents,
+    displayNames,
+    excludedCountries,
```

Comment:
not sure about removing the old trick, because here, your value as deps are always differents. `excludedCountries` is always recreated on every render, same for the rest of the deps :/

Reply:
See previous comment, I think TBH that all this "optimisation" can be removed. I don't think there is any issue in rerendering